### ISSUE TITLE
Subject keywords are disappearing when included in initial item deposit

### ISSUE THREAD

**Issue by paigecm** [DSpace/dspace-angular#2050]
**Describe the bug**
The standard deposit form includes a field for Subject Keywords. As of around January 18th, when you include keywords in the appropriate field, DSpace shows that they have been saved with the rest of the metadata, but when you complete the deposit, the keywords disappear, and cannot be seen, either in the default or full item view. I can reproduce this error in my local instance of DSpace, and at the [Lyrasis Demo repository](https://demo7.dspace.org/home)

If you go back and edit the item, then you can make the keywords appear. 

**To Reproduce**
1. Deposit an item following the normal process. Include subject keywords. Complete the deposit.
2. Look at the item view (both default and full view). The keywords will not show up.
3. Edit the item in order to add the keywords. They will not show up in the existing metadata, but you can re-add them.

**Expected behavior**
Keywords should be accepted and included in the initial deposit.

**Related work**
[Example of this bug in the demo repository](https://demo7.dspace.org/handle/10673/245277)

**Comment by tdonohue** [comment]
@paigecm : I'm not able to reproduce this in the demo site, or locally.  The demo site is _extremely slow_ today (likely a lot of people on it), but I was able to create a new Item with multiple subjects.  

Here's the item I created (it's in the same collection as the item you created): https://demo7.dspace.org/items/4148a524-9486-4628-afd7-aab4febe91c8

So, I'm not sure what could be going on.  Maybe the demo site was acting very slow when you tried this, and the "save" of these fields took longer?  I'm not exactly sure.  But, I also spun up pre-7.5 locally, and everything is working well there as well.

If you have more details on how to reproduce this reliably, or if you are able to find a way to reproduce this on *another site* (which is not the demo site), then please let me know.  I suspect right now that the slowness of the demo site caused things to not behave as expected.  (And we are working on migrating the demo site to a better environment, but it's unlikely to occur until Feb unfortunately.)

**Comment by paigecm** [comment]
Thanks, @tdonohue -- but that's weird, because I just reproduced it *again*, this time taking extra time to make sure that the system said it had saved the keywords. Here's the latest example, which should have the keyword "wombat": [https://demo7.dspace.org/handle/10673/245283](https://demo7.dspace.org/handle/10673/245283). 

I've also seen it in my local [instance](https://udspace.udel.edu/home), and tracked it back for several weeks for non-templated items.

I'll keep trying to figure out what's going on, but this is weird.

**Comment by tdonohue** [comment]
@paigecm : Oh, I think I understand what you may be doing now.  It sounds like there may be a usability issue here in the form field.

By default, the "Subject Keywords" field is configured to use a controlled vocabulary. Notice the "vocabulary" tag in this default config: https://github.com/DSpace/DSpace/blob/main/dspace/config/submission-forms.xml#L190-L201

This means you can only add keywords from this vocabulary, and as you type terms it will search that vocabulary. For instance, if you start typing "computer" you should see suggestions show up for "Computer Science" and similar.

If you *remove* that setting (comment out the "vocabulary" tag), then that allows the "Subject Keywords" field to accept any values.  In that scenario, you could add "wombat" or similar, and you'd comma-separate each keyword... that'd make them appear in the submission UI like this (I did this locally by commenting out the "vocabulary" tag on the backend config and restarting the backend):
![keywords-srsc-turned-off](https://user-images.githubusercontent.com/483997/213523105-f605d33a-2806-4aac-91d8-4373e9716c3a.png)

So, I think on the demo site, your keywords are not being accepted because they are not in the configured controlled vocabulary.  However, that looks like a usability issue to me because you likely should receive a notice or error message that the "keyword isn't found" in the vocabulary or similar.

Long story short, there is a small bug here where the UI isn't giving you an useful error when you attempt to add a term not in the controlled vocabulary.  But it also sounds like you may want to turn off this "vocabulary" configuration locally if you want the freedom to add any keyword that you want.

**Comment by paigecm** [comment]
I think that's it! Thanks so much, @tdonohue -- I'll work on getting that fixed locally. I can confirm I'm not seeing an error when I put in any old keyword, either in my local instance or the demo -- but getting that vocab tag commented out will fix my problem.

**Comment by vlguerrero** [comment]
I've found a similar issue wile trying to use tag input type without a controlled vocabulary on 7.4. Doing that, shows an error message (submission.sections.general.save_error_notice) and don't allow saving.

**Comment by tdonohue** [comment]
Closing as answered, since the originally reported issue was resolved.  Other related issues should be opened as separate tickets as needed.
